src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -663,6 +667,9 @@ void Aura::UpdateOwner(uint32 diff, WorldObject* owner)
 
     Update(diff, caster);
 
+    if (m_heartBeatTimer > 0)
```

Comment:
don't check timer, instead attribute (there's the slight possibility heartbeattimer reaches 0)

Reply:
how would it?

Reply:
timer = 0 is used as a cheap way to indicate "no timer needed" all over the core. this is fine.

Reply:
i would use `if (m_heartBeatTimer)` for readability/consistency with other uses